src/articulated_vehicles.cpp

```diff
@@ -80,20 +80,18 @@ uint CountArticulatedParts(EngineID engine_type, bool purchase_window)
 	 * either, so it doesn't matter how many articulated parts there are. */
 	if (!Vehicle::CanAllocateItem()) return 0;
 
-	Vehicle *v = nullptr;
+	std::unique_ptr<Vehicle> v;
 	if (!purchase_window) {
-		v = new Vehicle();
+		v.reset(new Vehicle());
```

Comment:
`v = std::make_unique<Vehicle>();`

Reply:
Actually now I'm not sure, because `new Vehicle()` allocates and deallocates via the pool, I'm not sure unique_ptr is appropriate here.

Reply:
Ah yes of course, silly me!

Reply:
`Vehicle::operator new` is inherited from `VehiclePool::PoolItem::operator new`.

Reply:
Ah okay, can confirm it does the [right thing](https://godbolt.org/z/hYxqdbqva).